### PR TITLE
Medical Statemachine - Fix locality EH breaking current state

### LINKED ISSUE #7126: Unconscious Issues (Not waking up, bugged wakeup)
We have been having rare but serious issues with units not waking up from unconscious
And when they do wakeup, they are sometimes bugged and cannot move.

Here is a recent example: 
```
19:26:14 (miscMedical) INFO: SLOG - uncon event: X [Unit:UNIT_BLU_TH2_P] [Active:true]
19:27:23 4x: [ACRE] (sys_server) WARNING: Desync - Garbage collection prevented for radio 'acre_prc148_id_3' from player 'X' as it still exists for them locally! idi\acre\addons\sys_server\fnc_stopRadioGarbageCollect.sqf:27
19:35:38 (miscMedical) INFO: SLOG - _checkUnconc: X [VitalsStableExamine: Stable: Should eventually wakeup] [Checker: Z] // external unit checks "hasStableVitals"
// I finaly notice him down from spectator and manually global exec 
// `if (profileName == "X") then { [player] call ace_medical_statemachine_fnc_handleStateUnconscious };`
19:43:48 (miscMedical) INFO: SLOG - wakeUp event: X [Unit:UNIT_BLU_TH2_P] [Stable: true]
19:43:53 (miscMedical) INFO: SLOG - wakeUp check: X [State:Injured] [isUncon:true] [Anim:unconscious] [Weap: rhs_weap_m3a1] [InputD: 105]
```
Even though he is unconscious and hasStableVitals there is no wakeUp event
I manually call handleStateUnconscious and immedietly get a wakeUp event
5 seconds later it checks and anim is broken and input blocker is still running
We got a `uncon-true` event, but never a `uncon-false`, yet is in state=Injured at the end

What could cause this? It seems like handleStateUnconscious was never called 
(probability of waking up is guaranteed with 10min waiting @ 0.5 chance)
Statemachine - statemachine not running for unit, or somehow not in correct state
LocalUnits - player somehow not in localUnits array
Possibly related to desync (acre warning)?

Medical Settings:
```
[QACEGVAR(medical,spontaneousWakeUpChance), 0.5], // default: 0.05
[QACEGVAR(medical_treatment,woundReopening), false], // default: true
[QACEGVAR(medical_treatment,litterCleanupDelay), 300], // default: 600
[QACEGVAR(medical_statemachine,AIUnconsciousness), false], // default: true
[QACEGVAR(medical_statemachine,cardiacArrestTime), 1], // default: 30
```

Debugging code:
```
["ace_medical_wakeUp", {
    params ["_unit"];
    if (_unit == ACE_player) then {
        private _stable = [_unit] call ace_medical_status_fnc_hasStableVitals;
        TRACE_2("wakeUp event",_unit,_stable);
        [QGVAR(sLog), ["wakeUp event", format ["%1 [Unit:%2] [Stable: %3]", profileName, _unit, _stable]]] call CBA_fnc_serverEvent;
        [{
            params ["_unit"];
            private _targetState = [_unit, ACEGVAR(medical,STATE_MACHINE)] call CBA_statemachine_fnc_getCurrentState;
            private _isUncon = _unit getVariable ["ACE_isUnconscious", -1];
            private _anim = animationState _unit;
            private _weap = currentWeapon _unit;
            private _inputD = missionNamespace getVariable ["ace_common_disableInputPFH", nil];
            if (isNil "_inputD") then {_inputD = "nil"};
            TRACE_6("wakeUp check",_unit,_targetState,_isUncon,_anim,_weap,_inputD);
            [QGVAR(sLog), ["wakeUp check", format ["%1 [State:%2] [isUncon:%3] [Anim:%4] [Weap: %5] [InputD: %6]", profileName, _targetState, _isUncon, _anim, _weap, _inputD]]] call CBA_fnc_serverEvent;
        }, [_unit], 5] call CBA_fnc_waitAndExecute;
    };
}] call CBA_fnc_addEventHandler;
["ace_unconscious", {
    params ["_unit", "_active"];
    if (_unit == ACE_player) then {
        TRACE_2("uncon event",_unit,_active);
        [QGVAR(sLog), ["uncon event", format ["%1 [Unit:%2] [Active:%3]", profileName, _unit, _active]]] call CBA_fnc_serverEvent;
    };
}] call CBA_fnc_addEventHandler;
```

I have tried but cannot reproduce this in small scale tests.

### PR DESCRIPTION
**Pull request by PabstMirror**
Fix #7126
Problems from PR https://github.com/acemod/ACE3/pull/6950

On locality change we have to do something to make sure state machine is in correct state
But I didn't realize how local EH actually works and each time a new client connected it was sending locality transfer events

This PR just uses the existing vars to determine state locally.

### CODE REVIEW

**Comment by commy2** [comment]
Can you explain how the Local event works with JIP etc?

**Comment by PabstMirror** [comment]
```
z1 (server):
localityChangedEH: _unit=z2, _isLocal=false

z2:
localityChangedEH: _unit=z2, _isLocal=false
localityChangedEH: _unit=z1, _isLocal=false
localityChangedEH: _unit=t1, _isLocal=false
2 frames later: localityChangedEH: _unit=z2, _isLocal=true
```

So anytime somone jipped, it was sending a event with the Default state

File: addons/medical_statemachine/functions/fnc_localityChangedEH.sqf [hunk 1]
```diff
@@ -19,15 +20,47 @@
...
+            if (_currentState == "Unconscious") exitWith {};
+            _unit setVariable [VAR_UNCON, false]; // force reset var so ace_medical_status_fnc_setUnconscious can run
+            TRACE_1("manually changing state to Unconscious",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Unconscious", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        // case (Injured?) - probably not needed, "Default" and "Injured" are the same
+        default {
+            // If locality transfers back and forth, we could be in an old state and should transfer back to default
+            if (_currentState == "Default") exitWith {};
+            if (_currentState == "Injured") exitWith {};
```
Comment (mharis001) [inline addons/medical_statemachine/functions/fnc_localityChangedEH.sqf]: Just a style thing, combine into one `if`?

File: addons/medical_statemachine/functions/fnc_localityChangedEH.sqf [hunk 2]
```diff
@@ -19,15 +20,47 @@
...
+            TRACE_1("manually changing state to CardiacArrest",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "CardiacArrest", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        case (IS_UNCONSCIOUS(_unit)): {
+            if (_currentState == "Unconscious") exitWith {};
+            _unit setVariable [VAR_UNCON, false]; // force reset var so ace_medical_status_fnc_setUnconscious can run
+            TRACE_1("manually changing state to Unconscious",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Unconscious", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        // case (Injured?) - probably not needed, "Default" and "Injured" are the same
```
Comment (mharis001) [inline addons/medical_statemachine/functions/fnc_localityChangedEH.sqf]: Should be fine, only thing would be the [FullHeal](https://github.com/acemod/ACE3/blob/master/addons/medical_statemachine/Statemachine.hpp#L30) event. The unit should go into Injured (from Default) if they are [bleeding or in pain](https://github.com/acemod/ACE3/blob/master/addons/medical_vitals/functions/fnc_handleUnitVitals.sqf#L152), but there could be a situation where neither is the case?

Comment (PabstMirror) [reply addons/medical_statemachine/functions/fnc_localityChangedEH.sqf]: Added injured check and now does transition

File: addons/medical_statemachine/functions/fnc_localityChangedEH.sqf [hunk 3]
```diff
@@ -19,15 +20,50 @@
...
+        default {
+            // If locality transfers back and forth, we could be in an old state and should transfer back to default
+            if (_currentState == "Default") exitWith {};
+            TRACE_1("manually changing state to Default",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Default", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+    };
+} else {
+    /*
+    // Not sure if this is even needed, idea is that on locality transfer we broadcast more up to date info
```
Comment (TheMagnetar) [inline addons/medical_statemachine/functions/fnc_localityChangedEH.sqf]: I think this should be the case in order to send the latest information, specially if the medical states vary between clients. In any case, it should be done before doing a manual transition to a particular state in my opinion.

**Pull request merged**